src/Microsoft.TestPlatform.Client/DesignMode/DesignModeTestHostLauncher.cs

```diff
@@ -30,15 +31,22 @@ public DesignModeTestHostLauncher(IDesignModeClient designModeClient)
     public virtual bool IsDebug => false;
 
     /// <inheritdoc/>
+    [Obsolete("Don't use, internal implementations should use only the latest interface.", true)]
     public bool AttachDebuggerToProcess(int pid)
     {
-        return _designModeClient.AttachDebuggerToProcess(pid, CancellationToken.None);
+        throw new NotSupportedException();
```

Comment:
Is it worth adding some more information in the exception message?

Reply:
Maybe. Let's see if we ever hit it, because I don't think we will.